cruft/_commands/link.py

```diff
@@ -43,6 +44,9 @@ def link(
             project_dir,
             checkout,
         )
+
+        validate_cookiecutter(cookiecutter_template_dir)
```

Comment:
Do the calls to `validate_cookiecutter` need to be in a try:except?

Reply:
I've moved this function out of `generate_cookiecutter_context` as it was validating the template too early once nesting was implemented (we want to validate only after the concrete template has been selected). To get around this I've moved the call a layer up the stack but other than that the functionality & error behaviour is the same.

It needs to be a terminating error, and cruft has pretty error handling (see screenshot) so I'm happy for it to live outside of a try/catch. We'd bubble the exception up to the user in any case
![image](https://github.com/user-attachments/assets/fbb8d54d-deb5-4cb9-a810-1da11c7b7f72)